resqpy/property.py

```diff
@@ -3039,6 +3039,332 @@ def basic_static_property_uuids_dict(self, realization = None, share_perm_parts
+      return self.collection.local_property_kind_uuid(self.part)
+
+   def facet_type(self):
+      """Returns string being the facet type for the property, or None if no facet present.
+
+      note: resqpy currently supports at most one facet per property, though RESQML allows for multiple.
+
+      :meta common:
+      """
+      return self.collection.facet_type_for_part(self.part)
+
+   def facet(self):
+      """Returns string being the facet value for the property, or None if no facet present.
+
+      note: resqpy currently supports at most one facet per property, though RESQML allows for multiple.
+
+      :meta common:
+      """
+      return self.collection.facet_for_part(self.part)
+
+   def time_series_uuid(self):
+      """Returns the uuid of the related time series (if applicable, otherwise None)."""
+      return self.collection.time_series_uuid_for_part(self.part)
+
+   def time_index(self):
+      """Returns int being the index into the related time series (if applicable, otherwise None)."""
+      return self.collection.time_index_for_part(self.part)
+
+   def minimum_value(self):
+      """Returns the minimum value for the property as stored in xml (float or int or None)."""
+      return self.collection.minimum_value_for_part(self.part)
+
+   def maximum_value(self):
+      """Returns the maximum value for the property as stored in xml (float or int or None)."""
+      return self.collection.maximum_value_for_part(self.part)
+
+   def uom(self):
+      """Returns string being the units of measure (for a continuous property, otherwise None).
+
+      :meta common:
+      """
+      return self.collection.uom_for_part(self.part)
+
+   def string_lookup_uuid(self):
+      """Returns the uuid of the related string lookup table (for a categorical property, otherwise None)."""
+      return self.collection.string_lookup_uuid_for_part(self.part)
+
+   def string_lookup(self):
+      """Returns a resqpy StringLookup table (for a categorical property, otherwise None)."""
+      return self.collection.string_lookup_for_part(self.part)
+
+   def constant_value(self):
+      """For a constant property, returns the constant value (float or int or bool, or None if not constant)."""
+      return self.collection.constant_value_for_part(self.part)
+
+   def from_array(self, cached_array, source_info, keyword,
+                  support_uuid, property_kind = None, local_property_kind_uuid = None,
+                  indexable_element = None, facet_type = None, facet = None,
+                  discrete = False, uom = None, null_value = None,
+                  time_series_uuid = None, time_index = None,
+                  realization = None, count = 1, const_value = None,
+                  string_lookup_uuid = None, find_local_property_kind = True, extra_metadata = {}):
+      """Populates this (empty) property from a numpy array and associated metadata.
```

Comment:
For consistency, this may make sense as a class constructor method, like from_singleton.
May be worth having this function located in the code next to the other constructor methods, for readability..